File: backend/app/schemas/document_queue.py
"""Document processing queue Pydantic schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.document_queue import DocumentQueueOperation, DocumentQueueStatus


class DocumentQueueItemResponse(BaseModel):
    """Response with document queue item details."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    document_id: UUID
    document_name: str  # Joined from documents table
    status: DocumentQueueStatus
    operation: DocumentQueueOperation
    attempts: int
    max_attempts: int
    error_message: str | None
    next_retry: datetime | None
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None


class DocumentQueueListResponse(BaseModel):
    """Paginated list of document queue items."""

    items: list[DocumentQueueItemResponse]
    total: int
    page: int
    page_size: int
    has_more: bool


class DocumentQueueStatsResponse(BaseModel):
    """Document queue statistics."""

    pending: int
    in_progress: int
    completed: int
    failed: int
    total: int

File: frontend/src/types/document-queue.ts
/**
 * Document processing queue types.
 */

/** Status for document queue items. */
export type DocumentQueueStatus = "pending" | "in_progress" | "completed" | "failed";

/** Operation type for document queue items. */
export type DocumentQueueOperation = "process" | "reprocess";

/** Document queue item from backend. */
export interface DocumentQueueItem {
  id: string;
  document_id: string;
  document_name: string;
  status: DocumentQueueStatus;
  operation: DocumentQueueOperation;
  attempts: number;
  max_attempts: number;
  error_message: string | null;
  next_retry: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
}

/** Paginated list of document queue items response. */
export interface DocumentQueueListResponse {
  items: DocumentQueueItem[];
  total: number;
  page: number;
  page_size: number;
  has_more: boolean;
}

/** Document queue statistics. */
export interface DocumentQueueStats {
  pending: number;
  in_progress: number;
  completed: number;
  failed: number;
  total: number;
}
